### PR TITLE
MUIC-410: [Android] Audio/Video – Send message triangle button disapp…

### PR DESCRIPTION
**Pull request by deniss-mc**
…ears, and user is not able to send messages

Send message rework
Cleanup

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java [hunk 1]
```diff
@@ -119,6 +119,33 @@ public void error(GliaException exception) {
...
+        }
+
+        @Override
+        public void errorOperatorNotOnline(String message) {
+            onSendMessageOperatorOffline(message);
+        }
+
+        @Override
+        public void errorMessageInvalid() {
+
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java:140]: can be removed if it is not used

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java [hunk 2]
```diff
@@ -148,11 +175,26 @@ public void error(GliaException exception) {
     private final GetFileAttachmentsUseCase getFileAttachmentsUseCase;
     private final RemoveFileAttachmentUseCase removeFileAttachmentUseCase;
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java:177]: empty space

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java [hunk 3]
```diff
@@ -810,11 +815,7 @@ public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2)
 
             @Override
             public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
-                if (charSequence.toString().trim().length() > 0) {
-                    sendButton.setVisibility(VISIBLE);
-                } else {
-                    sendButton.setVisibility(GONE);
-                }
+
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java:818]: extra line?

**Review by mustafa-ogun-ozturk — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java (hunk 3 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java:818]: linter is happy tho x) remains empty as the beforeTextChanged as i moved out the functionality

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java (hunk 1 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java:140]: usecase calls it, we might need to handle it when error handling will be introduced

Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java:140]: or at least by the tests :)

File: widgetssdk/src/main/java/com/glia/widgets/glia/GliaSendMessageUseCase.java [hunk 4]
```diff
@@ -6,68 +6,107 @@
...
+                listener.errorOperatorNotOnline(message);
+            }
+        } else {
+            listener.errorMessageInvalid();
+        }
     }
 
-    public void execute(String message, FilesAttachment filesAttachment, Listener listener) {
+    public void execute(SingleChoiceAttachment singleChoiceAttachment, Listener listener) {
         this.listener = listener;
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/glia/GliaSendMessageUseCase.java]: Didn't want to change too much of the initial approach - moved gliaeventlistener to repo

File: widgetssdk/src/main/java/com/glia/widgets/glia/GliaSendMessageUseCase.java [hunk 5]
```diff
@@ -6,68 +6,107 @@
...
 
-    public void execute(SingleChoiceAttachment singleChoiceAttachment, Listener listener) {
-        repository.sendMessage(singleChoiceAttachment, this);
+    private void sendMessage(String message) {
+        chatRepository.sendMessage(message, visitorMessageRequestCallback);
     }
 
-    public void execute(FilesAttachment filesAttachment, Listener listener) {
+    public void execute(String message, Listener listener) {
         this.listener = listener;
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/glia/GliaSendMessageUseCase.java]: Didn't want to change too much of the initial approach - moved gliaeventlistener to repo

**Pull request merged**